cmd/power/shutdown.go

```diff
@@ -14,31 +19,84 @@ func Main() {
 	app.Usage = "Control and configure RancherOS"
 	app.Version = config.VERSION
 	app.Author = "Rancher Labs, Inc."
-	app.Email = "sid@rancher.com"
 	app.EnableBashCompletion = true
 	app.Action = shutdown
+	app.HideHelp = true
 	app.Flags = []cli.Flag{
-		cli.StringFlag{
-			Name:  "r, R",
+		cli.BoolFlag{
+			Name:  "r",
 			Usage: "reboot after shutdown",
 		},
-		cli.StringFlag{
-			Name:  "h",
+		cli.BoolFlag{
+			Name:  "H",
 			Usage: "halt the system",
 		},
+		cli.BoolFlag{
+			Name:  "P",
+			Usage: "Power off the system",
+		},
 	}
-	app.HideHelp = true
-	app.Run(os.Args)
+	flags := []string{}
+	for _, arg := range os.Args {
+		if arg == "-h" {
+			flags = append(flags, "-H")
+		} else {
+			flags = append(flags, arg)
+		}
+	}
+	app.Run(flags)
 }
 
 func shutdown(c *cli.Context) {
+	timeToParse := ""
+	timeValue := c.Args().First()
+
+	if len(timeValue) == 0 || timeValue == "" {
+		log.Fatalf("No time value specified")
+	}
+
+	if timeValue[0] == '+' {
+		if len(timeValue) > 1 {
+			timeToParse = timeValue[1:]
+		} else {
+			log.Fatalf("invalid time value \"+\"")
+		}
+	} else if strings.Contains(timeValue, ":") {
+		reference := "15:04"
+		t, err := time.Parse(reference, timeValue)
+		if err != nil {
+			log.Fatalf("Could not parse time, err=%v", err)
+		}
+		now := time.Now()
+		y, m, d := now.Date()
+		a := t.AddDate(y, int(m)-1, d-1)
+		b := a.Add(time.Duration(int(now.Day()-a.Day())*24) * time.Hour)
+		mins := int(b.Sub(now).Minutes())
+		if mins < 0 {
+			mins = 0
+		}
+		timeToParse = strconv.Itoa(mins)
+	} else if timeValue == "now" {
+		timeToParse = "0"
+	} else {
+		log.Fatalf("invalid time value %s", timeValue)
+	}
 	common("")
-	reboot := c.String("r")
-	poweroff := c.String("h")
+	if os.ExpandEnv("${IN_DOCKER}") == "true" {
```

Comment:
Can you test if `ctrl-c` will cancel shutdown?  It should be if you run `shutdown -h +1` and then hit `ctrl-c` it won't shutdown.  I think you may need to sleep before `common("")`.